src/models/productDao.js

```diff
@@ -0,0 +1,338 @@
+const { DataSource } = require('typeorm');
+const path = require("path")
+const envFilePaht = path.resolve(__dirname, "../utils", ".env");
+const dotenv = require("dotenv")
+dotenv.config({ path: envFilePaht });
+
+const myDataSource = new DataSource({
+  type: process.env.TYPEORM_CONNECTION,
+  host: process.env.TYPEORM_HOST,
+  port: process.env.TYPEORM_PORT,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE
+})
+
+myDataSource.initialize()
+  .then(() => {
+    console.log("Data Source has been initialized!");
+  })
+  .catch((err) => {
+    console.error("Error occurred during Data Source initialization", err);
+    //   myDataSource.destroy();
+  });
+
+  
+// 유저 검증
+const foundUsers = async (id, email) => {
+  try {
+    const sql = await myDataSource.query(
+      `
+        SELECT id, email FROM users WHERE id = ? AND email = ?
+      `,[id, email]
+    );
+    return sql;
+  } catch (error) {
+    throw error;
+  }
+};
+// 해당 User의 장바구니 정보 호출
+const getUserCart= async (userid) => {
+  try {
+    const sql = await myDataSource.query( `
+      SELECT Products.id, Products.productImg, Products.name, Products.price, ShoppingItems.status, ShoppingItems.count, Products.quantity, Products.origin, Products.avm
+      FROM 
+        ShoppingItems 
+      JOIN 
+        Products ON ShoppingItems.productId = Products.id
+      WHERE ShoppingItems.userId = ?
+    `, [userid]);
+    return sql;
+  } catch(error) {
+    throw error;
+  }
+};
+// 상품 갯수를 파악 함
+const getItemsQuantity = async (id) => {
+  try {
+    const sql = await myDataSource.query(
+      `
+        SELECT quantity 
+          FROM  Products
+        WHERE id = ?
+      `, [id]
+    );
+    return sql;
+  } catch (error) {
+    throw error;
+  }
+};
+// Update SopphinItems Count
+const updateItemCount = async (productId, userId, count) => {
+  try {
+    const sql = await myDataSource.query(
+      `
+        UPDATE ShoppingItems 
+          SET count = ?, totalPrice = totalPrice + price
+        WHERE productId = ? AND userId = ?
+      `, [count, productId, userId]
+    );
+    return sql;
+  } catch (error) {
+    throw eorr
+  }
+};
```

Comment:
```suggestion
const updateItemCount = async (cartItemId, count) => {
  try {
    const sql = await myDataSource.query(
      `
        UPDATE ShoppingItems 
          SET count = ?, totalPrice = totalPrice + price
        WHERE productId = ? AND userId = ?
      `, [count, productId, userId]
    );
    return sql;
  } catch (error) {
    throw error
  }
};
```

가능하다면, 프론트로부터 productId와 userId를 각각 받아오는게 아니라, cartItemId 하나만 별도로 받아오시는 것이 좋을 것 같은데, 프론트 팀과 같이 논의한 번 해 주세요!